### PR TITLE
Capture sharing and printing times

### PR DESCRIPTION
**Pull request by jbethanco**
Replace ActivityViewController (poorly named) with ShareView that has
a callback which we can use to capture the last time the form was shared
or printed.

Capture printed date in printPDF function.

### CODE REVIEW

File: PBLogging/NewNavViews/PDFPreview/PDFPreviewView.swift [hunk 1]
```diff
@@ -74,7 +78,23 @@ struct PDFPreviewView: View {
...
+        if let error = error {
+            // TODO: Figure out what kind of errors can happen here and handle approprately vs this:
+            AlertProvider.shared.showAlertWithTitle(title: "Share error", message: "\(error), \(error.localizedDescription). Please screenshot and send to the dev team.")
+            return
+        }
+        if completed {
+            if activityType == UIActivity.ActivityType.print{
+                form.lastPrinted = Date()
+                NSLog("Form Last Printed: \(String(describing: form.lastPrinted))")
+            }else{
```
Comment (Pete-Hoch) [inline PBLogging/NewNavViews/PDFPreview/PDFPreviewView.swift:92]: Can we make this an explicit list of activities and then log a message when there is a non sharing activity.

For example is saving the PDF to local files sharing? It just feels like we can be listing things as shared that were not.

**Review by Pete-Hoch — CHANGES_REQUESTED** [review]
Can we also get the UIActivity Category? Then we can record a share for Category share and not action.

File: PBLogging/NewNavViews/PDFPreview/PDFPreviewView.swift (hunk 1 above)
Comment (jbethanco) [reply PBLogging/NewNavViews/PDFPreview/PDFPreviewView.swift:92]: We need to have a discussion on what is considered a share with Christian.
This PR gets the initial functionality in place. We can get more granular after feedback.

Comment (Pete-Hoch) [reply PBLogging/NewNavViews/PDFPreview/PDFPreviewView.swift:92]: The OS already makes a distinction between UIActivity.Category.share and UIActivity.Category.action. Can we use this?

**Review by Pete-Hoch — APPROVED** [review]
OK for now.

**Pull request merged**